src/API.ts

```diff
@@ -24,103 +21,73 @@ function getBackendURL(apiPath: string, path: string) {
+    },
+    data,
+    validateStatus: function (status) {
+      return status >= 200 && status < 300; // default
+    },
+  };
+
+  if (config.token) {
+    if (axiosConfig.headers) {
+      axiosConfig.headers['Authorization'] = `Bearer ${config.token}`;
+    }
   }
-  return response.body;
+
+  return axiosConfig;
 }
 
-export default class API {
-  [m: string]: (
-    path: string,
-    options: ApiRequestParams,
-  ) => Promise<request.Response>;
-
-  constructor(req?: Request) {
-    const cookies = new Cookies();
-
-    methods.forEach((method) => {
-      this[method] = async (
-        path,
-        {
-          config,
-          params,
-          data,
-          type,
-          headers = {},
-          checkUrl = false,
-        }: ApiRequestParams,
-      ) => {
-        let request: request.SuperAgentRequest;
-
-        // @ts-ignore
-        request = superagent[method](getBackendURL(config.apiPath, path));
-
-        if (params) {
-          request.query(params);
-        }
-        // let authToken;
-        // if (req) {
-        //   // @ts-ignore
-        //   // We are in SSR
-        //   authToken = req.universalCookies.get('auth_token');
-        //   request.use(addHeadersFactory(req));
-        // } else {
-        //   authToken = cookies.get('auth_token');
-        // }
-
-        if (config.token) {
-          request.set('Authorization', `Bearer ${config.token}`);
-        }
-
-        request.set('Accept', 'application/json');
-
-        if (type) {
-          request.type(type);
-        }
-
-        Object.keys(headers).forEach((key) => request.set(key, headers[key]));
-
-        if (data) {
-          request.send(data);
-        }
-
-        return request;
-
-        // request.end((err, response) => {
-        //   if (
-        //     checkUrl &&
-        //     request.url &&
-        //     request.xhr &&
-        //     stripQuerystring(request.url) !==
-        //       stripQuerystring(request.xhr.responseURL)
-        //   ) {
-        //     if (request.xhr.responseURL?.length === 0) {
-        //       return reject({
-        //         code: 408,
-        //         status: 408,
-        //         url: request.xhr.responseURL,
-        //       });
-        //     }
-        //     return reject({
-        //       code: 301,
-        //       url: request.xhr.responseURL,
-        //     });
-        //   }
-        //   return err ? reject(err) : resolve(response.body || response.text);
-        // });
-
-        // promise.request = request;
-        // return promise;
-      };
-    });
+export async function apiRequest(
+  method: string,
+  path: string,
+  options: ApiRequestParams,
+): Promise<any> {
+  const instance = axios.create();
+
+  if (options.raw) {
+    instance.interceptors.response.use(undefined, _handleError);
+  } else {
+    instance.interceptors.response.use(_handleResponse, _handleError);
   }
+
+  return instance.request(axiosConfigAdapter(method, path, options));
```

Comment:
Changed `instance()` to `instance.request()` to be able to mock easily in tests.